### DISCUSSION TITLE
Zniffer feature does not work with ACC-UZB-S

### DISCUSSION THREAD

**Discussion by mundschenk-at** [zwave-js/zwave-js#7036] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
I own an official (black) Sigma Design branded zniffer (ACC-UZB-S). Using the official Windows app, I can record wireless traffic but decoding S2 messages fails for me (copying the keys from zwave-js-ui). When trying to use the USB stick with the ZWave-JS zniffer feature, the stick does not work:

```
12:58:22.870 ZNIFFR   received Zniffer info:
                        Chip type:       ZW050x
                        Zniffer version: 2.43
2024-07-21 12:58:23.886 INFO ZNIFFERMANAGER: Retrying in 5s...
2024-07-21 12:58:24.189 DEBUG SOCKET: New connection <CONNECTIONSTRING>
2024-07-21 12:58:24.225 DEBUG SOCKET: Event INITED emitted to < CONNECTIONSTRING >
2024-07-21 12:58:24.233 DEBUG SOCKET: Event ZNIFFER_API emitted to < CONNECTIONSTRING>
2024-07-21 12:58:24.237 INFO APP: Zniffer api call: undefined
2024-07-21 12:58:28.905 ERROR ZNIFFERMANAGER: Zniffer error: Error Resource temporarily unavailable Cannot lock port
```

What did you expect to happen instead?
Being able to zniff traffic.

### Software versions

Driver (node-zwave-js):  13.0.1

Z-Wave JS UI:  9.16.1

If you are using something non-standard, tell us here: Currently trying to run on a standalone RPi 4 using NPM, but I previously also tried things in the HA add-on in a VM.

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [ ] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

No driver log created (because this instance is only running the Zniffer). There's nothing in the `zniffer_current.log` that I haven't included above.

[a comment by an automated account is not shown]

**Reply by mundschenk-at** [reply]
There's no driver log I can upload since the instance is not running a controller (and anyway, the issue is about the zniffer).

**Comment by AlCalzone** [comment]
> Resource temporarily unavailable Cannot lock port

usually means something else is using the same serial port at the same time.

**Reply by AlCalzone** [reply]
FWIW, I have the ACC-UZB**3**-S, not sure if that's the same as yours? Mine works.

**Reply by mundschenk-at** [reply]
I think the ACC-UZB3-S has a different SAW filter and double the amount of flash memory (according to https://www.silabs.com/documents/public/reference-manuals/INS12713-UZB-Product-Manual.pdf). 

Regarding the error message: I don't know how that could be, as I have tried this on three different systems, the last one being set up completely from scratch just for running the zniffer (fresh install of Raspberry Pi OS 64-bit).

**Reply by AlCalzone** [reply]
Hmm having the combined stdout of Z-Wave JS UI and the driver/Zniffer could help narrow it down. Just make sure to not log anything to file.

Maybe there's an error before that, which gets shadowed by the one you've shared.

**Reply by mundschenk-at** [reply]
```
$ STORE_DIR=~/.zwave-js-ui ZWAVEJS_EXTERNAL_CONFIG=~/.zwave-js-ui/.config-db zwave-js-ui
2024-07-24 19:40:07.847 INFO APP: Version: 9.16.1
2024-07-24 19:40:07.856 INFO APP: Application path:/usr/local/lib/node_modules/zwave-js-ui
  ______  __          __                      _  _____     _    _ _____ 
 |___  /  \ \        / /                     | |/ ____|   | |  | |_   _|
    / /____\ \  /\  / /_ ___   _____         | | (___     | |  | | | |  
   / /______\ \/  \/ / _' \ \ / / _ \    _   | |\___ \    | |  | | | |  
  / /__      \  /\  / (_| |\ V /  __/   | |__| |____) |   | |__| |_| |_ 
 /_____|      \/  \/ \__,_| \_/ \___|    \____/|_____/     \____/|_____|

2024-07-24 19:40:07.905 WARN STORE: scenes.json not found
2024-07-24 19:40:07.952 INFO APP: Listening on port 8091 protocol HTTP
2024-07-24 19:40:07.974 INFO ZNIFFERMANAGER: Initing Zniffer...
2024-07-24 19:40:07.986 INFO MQTT: MQTT is disabled
2024-07-24 19:40:07.989 WARN BACKUP: Store backup is disabled
2024-07-24 19:40:07.991 WARN BACKUP: Nvm backup is disabled
2024-07-24 19:40:07.995 INFO Z-WAVE: Z-Wave driver DISABLED
19:40:08.003 ZNIFFR   ███████╗ ███╗   ██╗ ██╗ ██████╗ ██████╗ ███████╗ ██████╗          ██╗ ███████╗
                      ╚══███╔╝ ████╗  ██║ ██║ ██╔═══╝ ██╔═══╝ ██╔════╝ ██╔══██╗         ██║ ██╔════╝
                        ███╔╝  ██╔██╗ ██║ ██║ ████╗   ████╗   █████╗   ██████╔╝         ██║ ███████╗
                       ███╔╝   ██║╚██╗██║ ██║ ██╔═╝   ██╔═╝   ██╔══╝   ██╔══██╗    ██   ██║ ╚════██║
                      ███████╗ ██║ ╚████║ ██║ ██║     ██║     ███████╗ ██║  ██║    ╚█████╔╝ ███████║
                      ╚══════╝ ╚═╝  ╚═══╝ ╚═╝ ╚═╝     ╚═╝     ╚══════╝ ╚═╝  ╚═╝     ╚════╝  ╚══════╝
19:40:08.010 SERIAL » 0xXXXXX                                                             (3 bytes)
19:40:08.021 SERIAL « 0xXXXXXXXX                                                     (7 bytes)
19:40:08.029 ZNIFFR   received Zniffer info:
                        Chip type:       ZW050x
                        Zniffer version: 2.43
19:40:08.032 SERIAL » 0xXXXXXX                                                          (4 bytes)
19:40:08.036 SERIAL « 0xXXXXX                                                             (3 bytes)
19:40:08.040 SERIAL » 0xXXXXX                                                             (3 bytes)
19:40:08.045 SERIAL « 0xXXXXXXXXXXXXXXXXXXXXXXXXX            (27 bytes)
19:40:08.049 SERIAL » 0xXXXXXx                                                           (4 bytes)
2024-07-24 19:40:08.568 DEBUG SOCKET: New connection <CONNECTIONSTRING>
2024-07-24 19:40:08.607 DEBUG SOCKET: Event INITED emitted to <CONNECTIONSTRING>
2024-07-24 19:40:09.053 INFO ZNIFFERMANAGER: Retrying in 5s...
2024-07-24 19:40:14.065 ERROR ZNIFFERMANAGER: Zniffer error: Error Resource temporarily unavailable Cannot lock port
Error: Error Resource temporarily unavailable Cannot lock port
2024-07-24 19:40:14.074 INFO ZNIFFERMANAGER: Retrying in 5s...
2024-07-24 19:40:19.085 ERROR ZNIFFERMANAGER: Zniffer error: Error Resource temporarily unavailable Cannot lock port
Error: Error Resource temporarily unavailable Cannot lock port
2024-07-24 19:40:19.088 INFO ZNIFFERMANAGER: Retrying in 5s...
2024-07-24 19:40:24.105 ERROR ZNIFFERMANAGER: Zniffer error: Error Resource temporarily unavailable Cannot lock port
Error: Error Resource temporarily unavailable Cannot lock port
```

I don't see much more (I XXXed out the SERIAL lines, not sure if there's anything potentially sensitive in there, probably not?).

**Reply by mundschenk-at** [reply]
When I remove the stick from the USB port, things fail in this way:

```
2024-07-24 19:44:56.857 ERROR ZNIFFERMANAGER: The serial port closed unexpectedly! (ZW0100)
ZWaveError: The serial port closed unexpectedly! (ZW0100)
    at SerialPort.onClose (/usr/local/lib/node_modules/zwave-js-ui/node_modules/@zwave-js/serial/src/ZnifferSerialPort.ts:40:10)
    at Object.onceWrapper (node:events:632:26)
    at SerialPort.emit (node:events:517:28)
    at port.close.then.closing (/usr/local/lib/node_modules/zwave-js-ui/node_modules/@serialport/stream/dist/index.js:242:18)
2024-07-24 19:44:59.751 ERROR ZNIFFERMANAGER: Zniffer error: Error: No such file or directory, cannot open /dev/serial/by-id/usb-0658_0200-if00
```

The VendorID and ProductID are the same as for my Aeotec Gen5+ stick, not sure why that is. No serial on either stick.

**Reply by AlCalzone** [reply]
It looks like it works at first and then it crashes because an expected response does not come. The serial data there should just be some initialization sequence, so you can share it. Or you can send it to me via email.

**Reply by mundschenk-at** [reply]
Sorry for the delay, I've sent you the logs via mail. I've experimented with various SiLabs COM drivers and Zniffer application versions have gotten the Windows Zniffer app to offer me a firmware update for the stick (from 2.43 to 2.55). However, there seems to be an issue with Auto Programming Mode not working and manually would apparently include shorting the `RST` pin to `GND` on the tiny PCB for the right amount of time, and I don't think I'll be able to do that :(.

**Reply by mundschenk-at** [reply]
So I took the plunge and flashed my Aeotec Gen5+ with the Zniffer firmware (2.55). That now works fine (got asked about the reset button again and that didn't seem to work correctly, even though the Aeotec has one, but after re-plugging the stick flashing went through without manual programming mode, no idea what the issue was the first time). 

Maybe the problem is specifically the Zniffer FW version?

**Comment by AlCalzone** [comment, marked as answer]
Escalated to an issue: https://github.com/zwave-js/node-zwave-js/issues/7052
